### PR TITLE
Comment rating

### PR DESCRIPTION
**Pull request by vvele017**
Something is in red in yarn.lock, but apart from yarn add react-rater, nothing else was added extra I believe. Contains the rate page and all components, buttons, that currently show. No button leads to the rate page yet, but to have files backed up in master as code that is good to go for now.

### CODE REVIEW

File: src/screens/book-ratings/Buttons/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+
+
+export default function CheckboxLabels(props:any) {
```
Comment (andres-valdes) [inline src/screens/book-ratings/Buttons/Checkbox.tsx]: arrow functions, no inline exports.

File: src/screens/book-ratings/Components/BookInfoComponent.css [hunk 2]
```diff
@@ -0,0 +1,119 @@
+
+p {
```
Comment (andres-valdes) [inline src/screens/book-ratings/Components/BookInfoComponent.css]: no .css, do all styling via material-ui

File: src/screens/book-ratings/Components/BookInfoComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    author: string,
+    userRating: number,
+    fivestar_total: number,
+    fourstar_total: number,
+    threestar_total: number,
+    twostar_total: number,
+    onestar_total: number
+}
+
+export default class BookInfoComponent extends React.Component
```
Comment (andres-valdes) [inline src/screens/book-ratings/Components/BookInfoComponent.tsx]: no class components, no inline exports.

File: src/screens/book-ratings/Components/CommentComponent.css [hunk 4]
```diff
@@ -0,0 +1,141 @@
+/*custom variables*/
```
Comment (andres-valdes) [inline src/screens/book-ratings/Components/CommentComponent.css]: no .css

File: src/screens/book-ratings/Components/CommentComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	commentboxDisabled: boolean,
+	nickname:			string,
+	ava:				any,
+	postRatingClicked:  boolean,
+	newComment:			any,
+};
+
+
+
+export default class CommentComponent extends React.Component
```
Comment (andres-valdes) [inline src/screens/book-ratings/Components/CommentComponent.tsx]: no class components, no inline exports.

File: src/screens/book-ratings/Items/CommentBox.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  }),
+);
+
+interface State {
+  maxWords: number,
+  minWords: number
+}
+
+
+export default function OutlinedTextFields(props:any) {
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/CommentBox.tsx]: no inline exports. use arrow functions

File: src/screens/book-ratings/Items/CommentBox.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+
+
+const useStyles = makeStyles((theme: Theme) =>
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/CommentBox.tsx]: useStyles below component.

File: src/index.tsx [hunk 8]
```diff
@@ -4,6 +4,11 @@ import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 
+import RatePage from './screens/book-ratings/RatePage'; //to be deleted after merged with book details
```
Comment (andres-valdes) [inline src/index.tsx]: don’t commit this.

File: src/screens/book-ratings/Items/LetterAvatar.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'; 
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import Avatar from '@material-ui/core/Avatar';
+import { indigo } from '@material-ui/core/colors';
+
+const useStyles = makeStyles(
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/LetterAvatar.tsx:16]: below component.

File: src/screens/book-ratings/Items/LetterAvatar.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  createStyles({
+    orangeAvatar: {
+      margin: 10,
+      color: '#fff',
+      backgroundColor: indigo[900],
+    },
+  }),
+);
+
+export default function LetterAvatar(props: any) {
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/LetterAvatar.tsx]: no export inline, use arrow functions

File: src/screens/book-ratings/Items/ProgressBar.tsx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+const TWOStarBar = withStyles({
+  root: {
+    height: 20,
+    width: 300, //width of the bar itself
+    backgroundColor: lighten('#C1C1C1', 0.5), //lighter background color of bar. keep same for all 
+  },
+  bar: {
+    borderRadius: 20,
+    //background: linear-gradient(angle, color-stop1, color-stop2);
+    background: 'linear-gradient(45deg, #FF8E53 30%, #ffcc00 90%)',
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/ProgressBar.tsx]: gradients don’t really fit the style.

File: src/screens/book-ratings/Items/UserComment.tsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    card: {
+      maxWidth: 450,
+      margin:  theme.spacing(2),
+    },
+    
+  }),
+);
+
+export default function CommentCard(props:any) {
```
Comment (andres-valdes) [inline src/screens/book-ratings/Items/UserComment.tsx]: no inline export, use arrow functions

File: src/screens/book-ratings/RatePage.css [hunk 13]
```diff
@@ -0,0 +1,43 @@
+body {
```
Comment (andres-valdes) [inline src/screens/book-ratings/RatePage.css]: no .css

File: src/screens/book-ratings/RatePage.tsx [hunk 14]
```diff
@@ -0,0 +1,43 @@
+/* Main Page for book rating
+   Will bring together all other components here
+*/
+import React from 'react';
+import './RatePage.css';
```
Comment (andres-valdes) [inline src/screens/book-ratings/RatePage.tsx]: dont use .css, use material-ui’s useStyle

File: src/serviceWorker.js [hunk 15]
```diff
@@ -0,0 +1,136 @@
+
```
Comment (andres-valdes) [inline src/serviceWorker.js]: what is this?

File: src/serviceWorker.ts [hunk 16]
```diff
@@ -1,143 +0,0 @@
-// This optional code is used to register a service worker.
```
Comment (andres-valdes) [inline src/serviceWorker.ts]: Why did you delete this?

**Review by andres-valdes — CHANGES_REQUESTED** [review]

File: src/serviceWorker.js (hunk 15 above)
Comment (vvele017) [reply src/serviceWorker.js]: idk i never went in to touch this myself tbh

Comment (andres-valdes) [reply src/serviceWorker.js]: Can you remove this and add the serviceWorker.ts that used to be here?

File: src/screens/book-ratings/Items/ProgressBar.tsx [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+              /> ;
+          break;
+  }
+
+  return (
+    <div className={classes.root}>{bar}</div>
+  );
+};
+
+const FIVEStarBar = withStyles({
```
Comment (mmarquez76) [inline src/screens/book-ratings/Items/ProgressBar.tsx:55]: This might be better/easier to read if you just made FIVEStarBar, FOURStarBar, etc their own styles under makeStyles()

**Review by mmarquez76 — APPROVED** [review]
Looks mostly good, only complaint is one minor change just to make your code more readable, but otherwise I'm fine merging this.

Waiting on @andres-valdes to check this out fully before merging, since he's the expert on this.

**Comment by mmarquez76** [comment]
Fixed merge conflict with package-lock.json, merging.
